substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -556,7 +556,7 @@ impl<'a, E: Ext, M: PolkaVmInstance<E::T>> Runtime<'a, E, M> {
 	}
 }
 
-impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
+impl<'a, E: Ext + 'a, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
```

Comment:
Merge error? This shouldn't be necessary?

Reply:
I think it is when this jobs run
```
SUBSTRATE_RUNTIME_TARGET=riscv cargo check -p minimal-template-runtime
```

Reply:
Curious. I guess due to the difference in toolchain version then.

Reply:
so it does now when we build 
```
SUBSTRATE_RUNTIME_TARGET=riscv cargo check -p minimal-template-runtime
```
because it pulls the umbrella crate that tries to build all the crates deps including pallet-revive to riscv.

Reply:
@kianenigma that explains why we are seeing all these compilations errors here and not in other PRs.
We could pick only the features we need in the Cargo.toml if we want to avoid that ...

Reply:
Hm, yea only selecting the needed dependencies should work.  
It's a bit annoying though, since the umbrella crate is supposed to make things easier. I guess it is still easier than selecting each dependency version manually.

Would it be possible to feature-gate the RISCV compilation?

Reply:
I gated the pallets that were causing issue, the long term solution is probably to generate the umbrella trait with feature flags that declare all their dependencies, so that you can hand pick the one you need and optimize your build time

Reply:
> yea only selecting the needed dependencies should work.
> It's a bit annoying though, since the umbrella crate is supposed to make things easier.

Couldn't we encode the dependency graph into the features in the script? Like you only activate the pallets you want and then they automatically activate the other pallets they depend on.

Reply:
I would not support making the usage of umbrella crate in any of the templates sub-optimal, for the sake of `SUBSTRATE_RUNTIME_TARGET=riscv` which is more of an experimental thing. 

All people out there will not face the same compilation issue, and people do use these templates as a source of truth, so it should be in the simplest form possible.